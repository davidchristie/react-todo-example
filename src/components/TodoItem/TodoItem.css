.TodoItem {
  box-shadow: 2px 2px 5px lightgray;  
  cursor: default;
  font-size: large;
  margin: 5px auto;
  padding: 10px;
  width: 80%;
}

.TodoItem.completed {
  color: lightgray;
  text-decoration: line-through;
}
