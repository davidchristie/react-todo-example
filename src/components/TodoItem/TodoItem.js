import PropTypes from 'prop-types'
import React from 'react'

import './TodoItem.css'

const TodoItem = props => {
  let className = 'TodoItem'
  if (props.item.completed) {
    className += ' completed'
  }
  return (
    <div className={className} onClick={props.onClick}>
      {props.item.text}
    </div>
  )
}

TodoItem.propTypes = {
  item: PropTypes.shape({
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired,
  onClick: PropTypes.func
}

export default TodoItem
