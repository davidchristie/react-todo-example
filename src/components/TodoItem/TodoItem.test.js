import { shallow } from 'enzyme'
import React from 'react'

import TodoItem from './TodoItem'

it('renders uncompleted items correctly', () => {
  const props = {
    item: {
      completed: false,
      text: 'Test todo item text'
    }
  }
  expect(shallow(<TodoItem {...props} />)).toMatchSnapshot()
})

it('renders completed items correctly', () => {
  const props = {
    item: {
      completed: true,
      text: 'Test todo item text'
    }
  }
  expect(shallow(<TodoItem {...props} />)).toMatchSnapshot()
})
