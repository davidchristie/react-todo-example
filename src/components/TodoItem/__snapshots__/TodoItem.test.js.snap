// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders completed items correctly 1`] = `
<div
  className="TodoItem completed"
>
  Test todo item text
</div>
`;

exports[`renders uncompleted items correctly 1`] = `
<div
  className="TodoItem"
>
  Test todo item text
</div>
`;
